### DISCUSSION TITLE
net80 -> net481/LangVersion 10: Unsafe.SkipInit

### DISCUSSION THREAD

**Discussion by thxkiwi** [microsoft/CsWin32#1191] [Q&A]
I have a project that I wrote using targeting framework net80 and trying to get it working with net481/LangVersion 10.

CsWin32: 0.3.106
Using Visual Studio 17.10.0.

I've modified the SDK project files to be 
```
  <PropertyGroup>
    <TargetFramework>net481</TargetFramework>
    <ImplicitUsings>enable</ImplicitUsings>
    <Nullable>enable</Nullable>
    <Platforms>x64</Platforms>
	  <LangVersion>10</LangVersion>
	  <EmitCompilerGeneratedFiles>true</EmitCompilerGeneratedFiles>
  </PropertyGroup>
```

CsWin32 is generating Windows.Win32.char.InlineArrays.g.cs:84 - `Unsafe.SkipInit(out __char_260 result);`
```
		public static unsafe implicit operator __char_260(ReadOnlySpan<char> value)
		{
			Unsafe.SkipInit(out __char_260 result);
			Span<char> target = new Span<char>(result.Value, SpanLength);
			value.CopyTo(target);
			int initLength = value.Length;
			target.Slice(initLength, SpanLength - initLength).Clear();
			return result;
		}
```

That seems to be .NET 8 only still.

Am I missing something (including "the point") here? It seemed possible and reasonable to target net481 and LangVersion 10. 

Help and guidance greatly appreciated.

**Comment by thxkiwi** [comment]
Needed an explicit selection of `System.Runtime.CompilerServices.Unsafe` package.

**Reply by AArnott** [reply]
Thanks for sharing. We shouldn't emit APIs that won't compile in your project, so this is a bug. Ideally CsWin32 would emit valid code for you and possibly emit a warning that suggests you add a package reference. I'll promote to a bug.
